.gitignore

```diff
@@ -1,6 +1,25 @@
-*.swp
-.DS_Store
-tmp
 bin/configlet
 bin/configlet.exe
-/.vagrant
+cabal.sandbox.config
+cabal-dev
+dist
+package.yaml
+stack.yaml
+tmp
+.DS_Store
```

Comment:
Should these be involved in Exercism?  Or is it better to educate contributors on where these go in `~/.gitignore`  Anything Haskell related I think should be here, but editor and OS are outside the scope of this project.